cmd/consume.go

```diff
@@ -54,8 +57,48 @@ func consume(messageType string, uri string, tenant string) error {
 	fmt.Println()
 
 	opts := make([]amqp.ConnOption, 0)
+
+	//Enable TLS
+	if insecure {
+		opts = append(opts, amqp.ConnTLSConfig(createTlsConfig()))
+	}else{
+		//Read cert from local TLS file 
+    	    	caCert, err := ioutil.ReadFile("tls.crt")       
```

Comment:
I think it makes sense adding this part ot the method `createTlsConfig`, so that other parts might re-use the same functionality.

I also think that the original use of the `insecure` flag was misleading here, as: insecure=false -> non-TLS, insecure=true -> TLS-insecure … but there is not "secure TLS" variant.

This might be a bit out of scope of your PR, but I would appreciate it, if you could fix this :-) … I think there should be a new `tls` flag, which replaces the original `insecure` behavior. And then let the `createTlsConfig` function deal with either secure or insecure.

Reply:
I completely Agree, See new commit.